cai_causal_graph/graph_components.py

```diff
@@ -110,6 +111,44 @@ def variable_type(self, new_type: Union[NodeVariableType, str]):
             raise TypeError(f'Expected NodeVariableType or string, got object of type {type(new_type)}.')
         self._variable_type = new_type
 
+    def _check_var_name_is_valid(self, name: str):
+        """Check that the name is valid."""
+        vname, _ = get_variable_name_and_lag(name)
+        # do the same for the current node
+        vname_this, _ = get_variable_name_and_lag(self.identifier)
```

Comment:
method comments and variables names are confusing, it's hard to tell what this code is doing